src/app/app.component.ts

```diff
@@ -94,10 +94,7 @@ export class AppComponent implements OnDestroy {
 			.pipe(takeUntil(this.unsubscribe$))
 			.subscribe((value) => {
 				if (value) {
-					this.dateRangeFormGroup.controls.dateRangeFormControl.patchValue(value, {
-						onlySelf: true,
-						emitEvent: false,
-					});
```

Comment:
Used to be an error that if you
- select 1y on date dropdown
- changed to date range picker on 3 weeks
- changed the dropdown to a lower value than the previous (so for example 1month)
then the `dateRangeFormControl` didn't get updated because of the `emitEvent: false` so you did not make a new API call to download chart data